src/main/java/org/red5/server/net/rtmp/RTMPConnection.java

```diff
@@ -122,19 +125,19 @@
 	 * 
 	 * @see org.red5.server.net.rtmp.Channel
 	 */
-	private transient ConcurrentMap<Integer, Channel> channels = new ConcurrentHashMap<Integer, Channel>(3, 0.9f, 1);
+	private transient ConcurrentMap<Integer, Channel> channels = new ConcurrentHashMap<Integer, Channel>(8, 0.9f, 4);
 

```

Comment:
It is working normal under load, but I agree this parameters should be configurable by developers. I can try to move this parameters into config in the new pull request, if you do not mind.

Yes, if the connection has not streams it uses only 3 channels. But even in this case we should increase concurrency level to permit parallel work with different channels.